packages/@n8n/task-runner/src/config/base-runner-config.ts

```diff
@@ -26,6 +26,14 @@ export class BaseRunnerConfig {
 	@Env('N8N_RUNNERS_MAX_CONCURRENCY')
 	maxConcurrency: number = 5;
 
+	/**
+	 * How long (in seconds) a runner may be idle for before exit. Intended
+	 * for use in `external` mode - launcher must pass the env var when launching
+	 * the runner. Disabled with `0` on `internal` mode.
+	 */
+	@Env('N8N_RUNNERS_IDLE_TIMEOUT')
```

Comment:
Maybe `N8N_RUNNERS_AUTO_SHUTDOWN_TIMEOUT` would be more explicit that it causes the runner to shutdown?